servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1656,9 +1745,8 @@ public void setState(TachyonURI path, SetAttributeOptions options)
    */
   public long scheduleAsyncPersistence(TachyonURI path)
       throws FileDoesNotExistException, InvalidPathException {
-    long workerId = scheduleAsyncPersistenceInternal(path);
-
     synchronized (mInodeTree) {
+      long workerId = scheduleAsyncPersistenceInternal(path);
```

Comment:
to avoid the need for nested synchronization (between mInodeTree and MWorkerToAsyncPersistFiles)